lib/dgram.js

```diff
@@ -344,6 +346,13 @@ Socket.prototype.send = function(buffer,
   if (typeof callback !== 'function')
     callback = undefined;
 
+  if (typeof address === 'function') {
+    callback = address;
+    address = undefined;
+  } else if (typeof address !== 'undefined' && typeof address !== 'string') {
+    throw new TypeError('"Address" argument must be a string or undefined');
```

Comment:
Even though it goes against the common error message formatting used in the rest of node core, I think changing this to `'Address must be a string or undefined'` might be better since it is more consistent with the error message format when validating the `port`.